src/audio/rtnr/rtnr.c

```diff
@@ -510,64 +514,53 @@ static int rtnr_reconfigure(struct comp_dev *dev)
 	return 0;
 }
 
-static int rtnr_set_comp_config_by_ipc_config_process(struct comp_dev *dev,
-							struct ipc_config_process *config)
+static int rtnr_set_config_data(struct comp_dev *dev,
+								unsigned char *data, uint32_t size)
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
 	int ret;
 
-	comp_dbg(dev, "rtnr_set_comp_config_by_ipc_config(): %d", config->size);
-	if (config->size != sizeof(cd->config)) {
-		comp_err(dev, "rtnr_set_comp_config_by_ipc_config(): invalid size %d",
-			config->size);
+	/* The initial data will be recevied as the combination of
+	 * config + data + switch. On some platforms only config is received
```

Comment:
I don't think the behavior is platform dependent. The variation should be brought from the age of working branch for platforms, or the caller: ipc_config_process & ipc_ctrl_data. The former comes along with TPLG_COMP_NEW command with combined blob, while the latter comes from COMP_SET_DATA.

Reply:
The comment is modified.